src/Http/Http.Extensions/src/HttpRequestJsonExtensions.cs

```diff
@@ -184,33 +141,32 @@ public static class HttpRequestJsonExtensions
     /// </summary>
     /// <param name="request">The request to read from.</param>
     /// <param name="type">The type of object to read.</param>
-    /// <param name="options">The serializer options use when deserializing the content.</param>
+    /// <param name="context">A metadata provider for serializable types.</param>
     /// <param name="cancellationToken">A <see cref="CancellationToken"/> used to cancel the operation.</param>
-    /// <returns>The task object representing the asynchronous operation.</returns>
-    [RequiresUnreferencedCode(RequiresUnreferencedCodeMessage)]
-    [RequiresDynamicCode(RequiresDynamicCodeMessage)]
+    /// <returns>The deserialized value.</returns>
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
     public static async ValueTask<object?> ReadFromJsonAsync(
+#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
         this HttpRequest request,
         Type type,
-        JsonSerializerOptions? options,
+        JsonSerializerContext context,
```

Comment:
Nit: Can we put the overloads back in their original order to make it easier to review?